src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -343,7 +342,8 @@ core::smart_refctd_ptr<ICPUBuffer> IShaderCompiler::CCache::serialize() const
     // Might as well memcpy everything
     memcpy(retVal.data() + SHADER_BUFFER_SIZE_BYTES + shaderBufferSize, dumpedContainerJson.data(), dumpedContainerJsonLength);
 
-    return core::make_smart_refctd_ptr<CVectorCPUBuffer<uint8_t, nbl::core::aligned_allocator<uint8_t>>>(std::move(retVal));
+    auto memoryResource = new core::VectorViewNullMemoryResource(std::move(retVal));
+    return ICPUBuffer::create({ .size = retVal.size(), .data = retVal.data(), .memoryResource = core::make_smart_refctd_ptr<core::refctd_memory_resource>(memoryResource) });
```

Comment:
after `retVal` gets moved you cannot call `.size` or `.data` anymore... they will return 0 and nullptr!

Reply:
you need to back up the values beforehand